util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -0,0 +1,458 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package worker implements the worker related activities like starting TCP/UDP/HTTP client/server
+and collecting the metric output to be returned to the controller when requested.
+*/
+package worker
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net"
+	"net/http"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/klog"
+)
+
+var listener net.Listener
+var resultStatus = make(chan string)
+var result []string
+var tstDuration string
+var tstProtocol string
+var startedAt int64
+var futureTime int64
+var iperfUDPFn = []string{"", "", "", "Sum", "Sum", "Sum", "Sum", "Sum", "Avg", "Avg", "Min", "Max", "Avg", "Sum"}
+var iperfTCPFn = []string{"Sum", "Avg"}
+
+const tcpMtrCnt = 2
+const udpMtrCnt = 11
+const httpMetrCnt = 12
+
+//StartResponse is the response sent back to controller on starting a measurement
+type startWrkResponse struct {
+	error string
+}
+
+//MetricResponse is the response sent back to controller after collecting measurement
```

Comment:
Comments should always be

// <type/function name> ...

Reply:
Modified.